src/clr-core/button/button.element.scss

```diff
@@ -1,6 +1,277 @@
+  letter-spacing: $clr-btn-appearance-standard-letter-spacing;
+  @include css-var(font-size, clr-btn-appearance-standard-font-size, $clr-btn-appearance-standard-font-size);
+  @include css-var(font-weight, clr-btn-appearance-standard-font-weight, $clr-btn-appearance-standard-font-weight);
+  @include css-var(height, clr-btn-appearance-standard-height, $clr-btn-appearance-standard-height);
+  @include css-var(padding, clr-btn-appearance-standard-padding, $clr-btn-appearance-standard-padding);
+}
+
+@mixin generateButton() {
+  @include btn-appearance();
+  @include btn-default-border-styles();
+  @include btn-min-max-widths();
+  @include btn-nowrap();
+  @include btn-text-properties();
+}
+
+@import '../../clr-angular/button/properties.button';
 
 :host {
-  color: blue;
+  box-sizing: border-box;
+
+  @include generateButton();
+  @include populateButtonProperties(default);
+
+  @include css-var(line-height, clr-btn-appearance-form-line-height, $clr-btn-appearance-form-line-height);
+  letter-spacing: $clr-btn-appearance-form-letter-spacing;
+  @include css-var(font-size, clr-btn-appearance-form-font-size, $clr-btn-appearance-form-font-size);
+  @include css-var(font-weight, clr-btn-appearance-form-font-weight, $clr-btn-appearance-form-font-weight);
+  @include css-var(height, clr-btn-appearance-form-height, $clr-btn-appearance-form-height);
+  @include css-var(padding, clr-btn-appearance-form-padding, $clr-btn-appearance-form-padding);
+
+  @include css-var(margin-top, clr-btn-vertical-margin, $clr-btn-vertical-margin);
+  @include css-var(margin-bottom, clr-btn-vertical-margin, $clr-btn-vertical-margin);
+  @include css-var(margin-right, clr-btn-horizontal-margin, $clr-btn-horizontal-margin);
+  margin-left: 0;
+}
+
+:host([icon]) {
+  min-width: 0;
+
+  &:host([disabled]) {
+    @include button-css-var(icon, color, disabled-color);
+  }
+}
+
+:host([block]) {
+  display: block;
+  width: 100%;
+  max-width: 100%;
+}
+
+:host([state='primary']) {
+  @include populateButtonProperties(primary);
+  @include btn-checked-styles(primary);
+}
+
+:host([state='success']) {
+  @include populateButtonProperties(success);
+  @include btn-checked-styles(success);
+
+  &:host([action='outline']) {
+    @include populateButtonProperties(success-outline);
+    @include btn-checked-styles(outline-success);
+  }
+}
+
+:host([state='danger']),
+:host([state='warning']) {
+  @include populateButtonProperties(danger);
+  @include btn-checked-styles(danger);
+
+  &:host([action='outline']) {
+    @include populateButtonProperties(danger-outline);
+    @include btn-checked-styles(outline-danger);
+  }
+}
+
+:host([action='outline']) {
+  @include populateButtonProperties(default-outline);
+}
+
+:host([state='inverse']) {
+  @include populateButtonProperties(inverse);
+}
+
+:host([action='link']) {
+  @include populateButtonProperties(link);
+  @include btn-checked-styles(link);
+  margin-right: 0;
+}
+
+:host([size='small']) {
```

Comment:
This is a small request with the API, could we make this `size=sm` that way we can same use t-shirt style size values for all the components? `sm`, `md`, `lg`